### PR TITLE
WIP [ENH,MAINT] HPI Coil Presets

### PR DESCRIPTION
**Pull request by gabrielbmotta**
- Add GUI for selecting HPI coil presets.
- Load presets from file.
- Some light refactoring to separate the parsing of digitizer data and displaying that data in the gui.

I ended up hijacking the existing json file that every mne scan plugin has and used it as the dev/enduser editable preset file. It gets coped to the plugin folder at build time.

### CODE REVIEW

File: libraries/disp/viewers/hpisettingsview.cpp [hunk 1]
```diff
@@ -416,11 +415,11 @@ void HpiSettingsView::onRemoveCoil()
 
 QList<FiffDigPoint> HpiSettingsView::readDigitizersFromFile(const QString& fileName)
 {
-    resetTables();
+    clearCoilGUI();
```
Comment (juangpc) [inline libraries/disp/viewers/hpisettingsview.cpp:418]: Better naming.

File: libraries/disp/viewers/hpisettingsview.cpp [hunk 2]
```diff
@@ -447,40 +446,9 @@ void HpiSettingsView::updateDigitizerInfo(const FiffDigPointSet& digSet)
     for(int i = 0; i < digSet.size(); ++i) {
         switch(digSet[i].kind) {
             case FIFFV_POINT_HPI: {
```
Comment (juangpc) [inline libraries/disp/viewers/hpisettingsview.cpp:448]: There is a strange set of { } after the case, with the break in there.

**Review by juangpc — DISMISSED** [review]

File: applications/mne_scan/plugins/hpi/hpi.cpp [hunk 3]
```diff
@@ -288,6 +290,9 @@ void Hpi::initPluginControlWidgets()
 
         // Projects Settings
         HpiSettingsView* pHpiSettingsView = new HpiSettingsView(QString("MNESCAN/%1/").arg(this->getName()));
```
[the remark this answers is not in the document]
Comment (gabrielbmotta) [reply applications/mne_scan/plugins/hpi/hpi.cpp:292]: Probably, but would involve changing how all plugin views are handled (in the AbstractPlugin class). If we do that, I think it should be its own pull request.

[the remark this answers is not in the document]
Comment (juangpc) [reply applications/mne_scan/plugins/hpi/hpi.cpp:292]: Yep. Focus.

**Pull request merged**